eox_nelp/pearson_vue/api/v1/views.py

```diff
@@ -130,9 +131,26 @@ class ResultNotificationView(PearsonRTENBaseView):
     This view handles the creation of Result Notification events in the Pearson RTEN system.
     The `event_type` attribute is set to "resultNotification".
     """
-
     event_type = RESULT_NOTIFICATION
 
+    def create(self, request, *args, **kwargs):
+        """
+        Execute the parent create method and allow to run the result notification pipeline.
+
+        Args:
+            request (Request): The request object containing the data.
+
+        Returns:
+            Response: Response object with status code 201 and an empty dictionary.
+        """
+        response = super().create(request, *args, **kwargs)
+
+        if response.status_code == status.HTTP_200_OK and getattr(settings, "ENABLE_CERTIFICATE_PUBLISHER", False):
+            result_notification = ResultNotificationBackend(request_data=request.data)
+            result_notification.run_pipeline()
```

Comment:
This pipeline is run in sync. Why not async?

Reply:
If fails, the record would not be created in rten model

Reply:
Let me answer with a question, Why async ?

Reply:
> If fails, the record would not be created in rten model

yes, probably the generics drf views implement an atomic block, but the result  will be stored if that fails due to pearson exceptions that are handled by the ErrorRealTimeImportHandler class

Reply:
> Let me answer with a question, Why async ?

Because in that case you isolate the RTEN request process from the generated certificate process.
You could avoid some 500 error codes, and maybe the client(Pearson) would think why we are getting 500...
All code related to the pipeline running in sync could make the request slower or fails and I think the Pearson Client is not aware that it is not related with the purpose of data sent...

Reply:
Errors should never pass silently. Python zen 
To avoid errors I can use a simple try except but I want that the client get a  500 error is the only way to know that something is happening 
request slower, yes, every line that you add in code affects the time response and when the code that you add is too much or needs a lot of time to be executed , one option is an async task, so is this that case ? have you experimented a slow response ? or have you identified a pipeline step that needs a lot of time ?

Reply:
errors would happen in the sync or async context. Anyway sentry would report both envs. So it wouldn't pass silently meanwhile sentry is working xD.
The point is who reports the error: the worker or the lms pod. And also who is being affected by that...

Anyway, if you consider is ok I could accept that.

Reply:
The issue isn't about what reports the errors; it's about who receives them. On one hand, we have the client using the service, and on the other, we have the Sentry service that goes unchecked until the client complains.

Finally, t main purpose of the async tasks is to execute heavy processes, not to avoid exceptions. For that, you can catch the exceptions and use a simple log.error if you want to report it to Sentry.